plugins/inventory/lxd.py

```diff
@@ -0,0 +1,1090 @@
+            - Specify C(inet) for IPv4 and C(inet6) for IPv6.
+            required: false
+            type: str
+            default: inet
+            choices: [ 'inet', 'inet6' ]
+        groups:
+            description:
+            - Create Groups by the following keywords location, pattern, network_range, os, release, profile, vlanid.
+            - See example for syntax.
+            required: false
+            type: json
+            default: none
+        dumpdata:
+            description: Dump out data to debug.
+            required: false
+            type: bool
+            default: false
+'''
+
+EXAMPLES = '''
+# simple lxd.yml
+plugin: community.general.lxd
+url: unix:/var/snap/lxd/common/lxd/unix.socket
+dumpdata: false
+
+# simple lxd.yml including filter
+plugin: community.general.lxd
+url: unix:/var/snap/lxd/common/lxd/unix.socket
+state: RUNNING
+dumpdata: false
+
+# grouping lxd.yml
+groups:
+  testpattern:
+    type: pattern
+    attribute: test
+  vlan666:
+    type: vlanid
+    attribute: 666
+  locationBerlin:
+    type: location
+    attribute: Berlin
+  osUbuntu:
+    type: os
+    attribute: ubuntu
+  releaseFocal:
+    type: release
+    attribute: focal
+  releaseBionic:
+    type: release
+    attribute: bionic
+  profileDefault:
+    type: profile
+    attribute: default
+  profileX11:
+    type: profile
+    attribute: x11
+  netRangeIPv4:
+    type: network_range
+    attribute: 10.98.143.0/24
+  netRangeIPv6:
+    type: network_range
+    attribute: fd42:bd00:7b11:2167:216:3eff::/24
+dumpdata: false
+'''
+
+import binascii
+import json
+import re
+import time
+import os
+import socket
+from ansible.plugins.inventory import BaseInventoryPlugin
+from ansible.module_utils._text import to_native
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible_collections.community.general.plugins.module_utils.lxd import LXDClient, LXDClientException
+
+
+class InventoryModule(BaseInventoryPlugin):
+    DEBUG = 4
+    NAME = 'community.general.lxd'
+    SNAP_SOCKET_URL = 'unix:/var/snap/lxd/common/lxd/unix.socket'
+    SOCKET_URL = 'unix:/var/lib/lxd/unix.socket'
+    SELFTEST = False
+
+    @staticmethod
+    def load_json_data(path):
+        """Load json data
+
+        Load json data from file
+
+        Args:
+            list(path): Path elements
+            str(file_name): Filename of data
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            dict(json_data): json data"""
+        try:
+            with open(path, 'r') as json_file:
+                return json.load(json_file)
+        except IOError as err:
+            raise AnsibleParserError('Could not load the test data: {0}'.format(to_native(err)))
+        except json.decoder.JSONDecodeError as err:
+            raise AnsibleParserError('The format of the JSON file {0} is incorrect: {1}'.format(to_native(path), to_native(err)))
```

Comment:
hallo @conloos , given that the json exception is probably going to be clear about format problems, this might be simplified to:
```suggestion
        except (IOError, json.decoder.JSONDecodeError) as err:
            raise AnsibleParserError('Could not load the test data from {0}: {1}'.format(to_native(path), to_native(err)))
```

Reply:
done